### PR TITLE
Interval Documentation

### PR DESCRIPTION
**Pull request by bbrennanbasis**
Pull request to publish Interval documentation.

### CODE REVIEW

File: docs/advanced_topics/interval.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import JavadocLink from '@site/src/components/DocsTools/JavadocLink';
+import DocChip from '@site/src/components/DocsTools/DocChip';
+
+<JavadocLink type="foundation" location="com/webforj/Interval" top='true'/>
+
+## Overview
+The `Interval` class represents a timer that triggers an [event](../ui/events) with a fixed time delay between each triggering.
+
+The `Interval` class provides a straightforward way to trigger events after a specified delay. It's possible to start, stop, and restart an `Interval` as needed. Additionally, Intervals can support multiple listeners for the elapsed event.
+Optimized for the webforJ framework, it offers better performance compared to the standard Java timer or the Swing timer.
```
Comment (MatthewHawkins) [inline docs/advanced_topics/interval.md:15]: We'll want to check with Hyyan to ensure this is correct, but leave it for his review

File: docs/advanced_topics/interval.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+5. **Live Data**: Refresh data on your app, such as stock prices, news feeds, or weather updates, on each Interval to keep the data current.
+
+6. **Interacitve Tutorials**: Create step-by-step tutorials where each step or instruction appears at a timed interval, guiding users through a process or feature of your website.
+
+
+## Constructor
+<JavadocLink type="foundation" location="com/webforj/Interval" code='true' suffix='#<init>(float,com.webforj.dispatcher.EventListener)'>Interval(float delay, EventListener\<Interval.ElapsedEvent\> listener)</JavadocLink>: Creates an `Interval` that triggers the given [event](../ui/events) with the specified delay as a float value.
+
+:::tip
```
Comment (MatthewHawkins) [inline docs/advanced_topics/interval.md:44]: I'd move this tip to the delay section

**Review by MatthewHawkins — CHANGES_REQUESTED** [review]
Looks good Ben! Go ahead and address the small changes I commented on and we'll pass this along. Once you've made the changes, please refresh/re-request a review.

File: docs/advanced_topics/interval.md (hunk 1 above)
Comment (bbrennanbasis) [reply docs/advanced_topics/interval.md:15]: I got the introduction information from the [Interval JavaDoc](https://javadoc.io/doc/com.webforj/webforj-foundation/latest/com/webforj/Interval.html), so if it is incorrect we'll want to change it there as well.

File: docs/advanced_topics/interval.md (hunk 2 above)
Comment (bbrennanbasis) [reply docs/advanced_topics/interval.md:44]: To clarify, do you mean the **Changing the Delay** section?

Comment (MatthewHawkins) [reply docs/advanced_topics/interval.md:44]: Correct :)

File: docs/advanced_topics/interval.md (hunk 1 above)
Comment (MatthewHawkins) [reply docs/advanced_topics/interval.md:15]: Okay perfect - just wanted to make sure it was coming from our docs

**Review by MatthewHawkins — DISMISSED** [review]

File: docs/advanced_topics/interval.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  });
+```
+
+## Changing the Delay
+To change the delay of the Interval, use the `setDelay(float delay)` method. The new float value takes effect once the Interval has stopped or restarted. Here's how it was set up for the demo at the top of the page:
+
+```java
+//Changing the Delay
+  restartInterval.addClickListener(e -> {
+    //... Grab 'pickedSpeed', the number from the NumberField as float value
```
Comment (hyyan) [inline docs/advanced_topics/interval.md:73]: what this comment means ?

**Review by hyyan — CHANGES_REQUESTED** [review]

File: docs/advanced_topics/interval.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+---
+sidebar_position: 45
+title: Interval
+---
+import ComponentDemo from '@site/src/components/DocsTools/ComponentDemo';
+import JavadocLink from '@site/src/components/DocsTools/JavadocLink';
+import DocChip from '@site/src/components/DocsTools/DocChip';
+
+<JavadocLink type="foundation" location="com/webforj/Interval" top='true'/>
```
Comment (hyyan) [inline docs/advanced_topics/interval.md:9]: Remove this link

File: docs/advanced_topics/interval.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+The `Interval` class triggers event(s) at a fixed time delay. Leveraging Intervals creatively, you can enhance user interaction and engagement on your website while keeping the experience dynamic and interesting.:
+
+1. **Check for Inactivity**: Display a [`Dialog`](../components/dialog) component if there hasn't been any interaction on a form within a given time.
+
+2. **Featured Content**: Rotate through featured articles, products, or promotions on your homepage on each Interval. This keeps the content dynamic and engaging.
+
+3. **Live Data**: Refresh data on your app, such as stock prices, news feeds, or weather updates, on each Interval to keep the data current.
+
+## Managing Interval States: Starting, Stopping, and Restart
+ use the start() method to initiate it. To halt the Interval, employ the stop() method. The restart() method can be used to restart the Interval.
```
Comment (hyyan) [inline docs/advanced_topics/interval.md:27]: This a duplication of the line after ?

**Review by hyyan — CHANGES_REQUESTED** [review]

**Comment by hyyan** [comment]
@bbrennanbasis  looks good, thanks. I asked for few minor changes

File: docs/advanced_topics/interval.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+## Overview
+The <JavadocLink type="foundation" location="com/webforj/Interval" code='true' >Interval</JavadocLink> class represents a timer that triggers an [event](../ui/events) with a fixed time delay between each triggering.
+
+The `Interval` class provides a straightforward way to trigger events after a specified delay. It's possible to start, stop, and restart an `Interval` as needed. Additionally, Intervals can support multiple listeners for the elapsed event.
+Optimized for the webforJ framework, it offers better performance compared to the standard Java timer or the Swing timer.
+
+## Usages
+The `Interval` class triggers event(s) at a fixed time delay. Leveraging Intervals creatively, you can enhance user interaction and engagement on your website while keeping the experience dynamic and interesting.:
+
+1. **Check for Inactivity**: Display a [`Dialog`](../components/dialog) component if there hasn't been any interaction on a form within a given time.
```
Comment (hyyan) [inline docs/advanced_topics/interval.md:17]: Dialog link is broken

Comment (bbrennanbasis) [reply docs/advanced_topics/interval.md]: When testing on my local device, the link correctly went to Dialog, but an unrelated error popped up for every component page that uses the table builder.

Comment (hyyan) [reply docs/advanced_topics/interval.md]: @MatthewHawkins could please check this issue with @bbrennanbasis

Comment (MatthewHawkins) [reply docs/advanced_topics/interval.md]: Sure, will review in Thursday's meeting

Comment (MatthewHawkins) [reply docs/advanced_topics/interval.md]: Checked the issue on the current branch and it works on dev version - should be fine on live.

**Review by hyyan — APPROVED** [review]

**Pull request merged**